### PR TITLE
READY ~ Rivtech magazine update.

### PR DESCRIPTION
**Pull request by Rivet-the-Zombie**
Adds magazine support to and updates Rivtech weapons. Updates No Fictional Weapons mod with new content.

---

Known issue: The RM121 aux shotgun can't detect/load its magazine, but that's outside the scope of this set of updates.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Ah, nice. Profession tweaks seem like a logical idea, though I wonder if there's a way to set a profession to start with pre-loaded magazines.

**Comment by mugling** [comment]
> The RM121 aux shotgun can't detect/load its magazine, but that's outside the scope of this set of updates

Can you walk me through the steps and I'll look into this

File: data/json/items/ranged.json [hunk 1]
```diff
@@ -3814,28 +3815,30 @@
...
     "durability": 9,
     "burst": 5,
-    "clip_size": 50,
-    "reload": 500,
+    "reload": 100,
     "loudness" : 150,
-    "valid_mod_locations": [[ "accessories", 4 ],[ "barrel", 1 ],[ "conversion", 1 ],[ "grip", 1 ],[ "magazine", 2 ],[ "mechanism", 4 ],[ "muzzle", 1 ],[ "rail", 1 ],[ "sights", 1 ],[ "stock", 1 ],[ "underbarrel", 1 ]]
+    "magazines" : ["8x40_50_mag", "8x40_100_mag"],
+    "magazine_well" : 2,
+    "default_mods": [ "8x40_50_mag" ],
```
Comment (mugling) [inline data/json/items/ranged.json]: Perhaps avoid `default_mods` whilst we work on better magazine spawns

Comment (Rivet-the-Zombie) [inline data/json/items/ranged.json]: Do we have any sort of alternative at the moment, or would they be spawning in without magazines?

**Comment by Rivet-the-Zombie** [comment]
>Can you walk me through the steps and I'll look into this

Attach aux shotgun to rifle, switch to aux shotgun firing mode, Use **U**nload command to remove magazine from aux shotgun, load magazine, attempt to load magazine into aux shotgun by reloading aux shotgun.

It fails to detect the aux shotgun magazine when it comes to the final step.

**Comment by chaosvolt** [comment]
Auxiliary weapons have always been borked in many special ways. For example, last I checked attaching a brass catcher and shotgun suppressor doesn't affect the secondary function of the KSG.

Then again, that was a very hackish way to represent a second tub magazine in the first place, so an increased number of inconsistencies are inevitable.

**Comment by Rivet-the-Zombie** [comment]
Closing due to  #15072.

**Comment by Rivet-the-Zombie** [comment]
My mistake; this one should remain open.

**Comment by mugling** [comment]
As of d099de3 `COMPACT_MAG` has become `MAG_COMPACT` and `BULKY_MAG` has become `MAG_BULKY`. See #15072 for explanation.

**Comment by Rivet-the-Zombie** [comment]
Updated.

**Comment by Rivet-the-Zombie** [comment]
Updated with new flags.

**Comment by Rivet-the-Zombie** [comment]
Updated with new flags.

**Pull request merged**